packages/table/src/Table.ts

```diff
@@ -348,6 +363,22 @@ export class Table extends SizedMixin(SpectrumElement, {
                 row.insertAdjacentElement('afterbegin', checkbox);
                 row.selected = this.selectedSet.has(row.value);
                 checkbox.checked = row.selected;
+
+                // if we have a rowheader, then update the label of the checkbox
+                // Clients can specify a selectRowLabel attribute on the rowheader that will
+                // be used as the label for the checkbox. If not present, the default
+                // selectRowLabel will be used.
+                const rowHeader = row.querySelector('[role="rowheader"]');
+                if (rowHeader) {
+                    const cell = rowHeader as TableCell;
+                    checkbox.ariaLabel = cell.selectRowLabel
+                        ? cell.selectRowLabel
+                        : `${
+                              this.selectRowLabel
+                          } ${rowHeader.textContent?.trim()}`;
```

Comment:
Did we add a unit test for when `textContent` is undefined?

Reply:
not yet. I will add a unit test. Thanks